src/udev/udevadm-trigger.c

```diff
@@ -152,6 +153,29 @@ static int device_monitor_handler(sd_device_monitor *m, sd_device *dev, void *us
         }
 
         settle_id = hashmap_get2(settle_hashmap, syspath, (void**) &k);
+        if (!settle_id) {
+                const char *old_sysname;
+
+                /* When the device is renamed, the new name is broadcast, and the old name is saved in INTERFACE_OLD. */
+
+                if (sd_device_get_property_value(dev, "INTERFACE_OLD", &old_sysname) >= 0) {
```

Comment:
hmm, this special casing should probably only done for network devices? i.e. needs a SUBSYSTEM check first?